tool.ts

```diff
@@ -402,20 +403,24 @@ export async function extract7z(file: string, dest?: string, _7zPath?: string):
 
         if (_7zPath) {
             // extract
-            let _7z: trm.ToolRunner = tl.tool(_7zPath)
-                .arg('x')         // eXtract files with full paths
-                .arg('-bb1')      // -bb[0-3] : set output log level
-                .arg('-bd')       // disable progress indicator
-                .arg('-sccUTF-8') // set charset for for console input/output
-                .arg(file);
+            let _7z: trm.ToolRunner = tl.tool(_7zPath);
+            if (overwriteDest) {
+                _7z.arg('-aoa');
+            }
+
+            _7z.arg('x')         // eXtract files with full paths
+               .arg('-bb1')      // -bb[0-3] : set output log level
+               .arg('-bd')       // disable progress indicator
+               .arg('-sccUTF-8') // set charset for for console input/output
+               .arg(file);
             await _7z.exec();
         }
         else {
             // extract
             let escapedScript = path.join(__dirname, 'Invoke-7zdec.ps1').replace(/'/g, "''").replace(/"|\n|\r/g, ''); // double-up single quotes, remove double quotes and newlines
             let escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, '');
             let escapedTarget = dest.replace(/'/g, "''").replace(/"|\n|\r/g, '');
-            let command: string = `& '${escapedScript}' -Source '${escapedFile}' -Target '${escapedTarget}'`
+            let command: string = `& '${escapedScript}' -Source '${escapedFile}' -Target '${escapedTarget}' -OverrideDestDirectory ${overwriteDest ? 1 : 0}`;
```

Comment:
I am passing this boolean as a number to the PowerShell script since Powershell is trying to cast true/false values to the string instead of boolean, which causes typing errors in the script. Numbers are correctly parsed as boolean in the script, so I am using them here.